### PR TITLE
[terra-action-header] Add support to allow users to set A11y label to action buttons.

### PR DESCRIPTION
**Pull request by saket2403**
### Summary

Closes #

### Deployment Link

https://terra-core-deployed-pr-#.herokuapp.com/

### Additional Details

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-action-header/src/ActionHeader.jsx
Comment (supreethmr) [inline packages/terra-action-header/src/ActionHeader.jsx:16]:
```suggestion
  backButtonA11yLabel: PropTypes.string,
```

Comment (saket2403) [reply packages/terra-action-header/src/ActionHeader.jsx]: Done [c3fd549](https://github.com/cerner/terra-core/pull/3645/commits/c3fd5495440c65ccc397d1e1c90290c255ceab0a)

File: packages/terra-action-header/src/ActionHeader.jsx
Comment (benbcai) [inline packages/terra-action-header/src/ActionHeader.jsx:34]:
```suggestion
   * Accessibility label for Next button. To be used with onNext prop
```

File: packages/terra-action-header/src/ActionHeader.jsx
Comment (benbcai) [inline packages/terra-action-header/src/ActionHeader.jsx:73]:
```suggestion
   * Accessibility label for Previous button. To be used with onPrevious prop.
```

File: packages/terra-action-header/src/ActionHeader.jsx
Comment (benbcai) [inline packages/terra-action-header/src/ActionHeader.jsx:118]: Should these text be translated?

Comment (saket2403) [reply packages/terra-action-header/src/ActionHeader.jsx:118]: Yes, I have asked for Neil's approval for the text and will log the translation JIRA and replace these.

File: packages/terra-action-header/src/ActionHeader.jsx
Comment (supreethmr) [inline packages/terra-action-header/src/ActionHeader.jsx:124]: don't we need similar descriptive texts for Back button, Next button and Previous button.

**Comment by supreethmr** [comment]
It would be nice if we have some examples updated with the usage of this new props.

File: packages/terra-action-header/src/ActionHeader.jsx
Comment (saket2403) [reply packages/terra-action-header/src/ActionHeader.jsx:124]: The descriptive texts will be passed as props for these Back, Next and Prev buttons to icon Button's text prop. Since we are not sure of the context in which these buttons will be used.

File: packages/terra-action-header/src/ActionHeader.jsx
Comment (supreethmr) [inline packages/terra-action-header/src/ActionHeader.jsx:118]: Lets use same Id for all.
```suggestion
  const buttonId = uniqueid();
  const closeButtonId = `terra-action-header-close-button-${buttonId}`;
  const maximizeButtonId = `terra-action-header-maximize-button-${buttonId}`;
  const minimizeButtonId = `terra-action-header-minimize-button-${buttonId}`;
```

File: packages/terra-action-header/src/ActionHeader.jsx
Comment (supreethmr) [inline packages/terra-action-header/src/ActionHeader.jsx:122]: Is `aria-hidden` added for any specific screen reader and browser combination..??

Comment (saket2403) [reply packages/terra-action-header/src/ActionHeader.jsx:122]: Without aria-hidden screen reader will read these text twice

**Review by supreethmr — APPROVED** [review]

File: packages/terra-action-header/src/ActionHeader.jsx
Comment (benbcai) [inline packages/terra-action-header/src/ActionHeader.jsx:237]: Is this what is causing the title in the action header to shift to the left in the screenshots. It looks like there isn't much left padding in the new screenshots. Is this expected?

File: packages/terra-action-header/src/ActionHeader.jsx
Comment (xenoworf) [inline packages/terra-action-header/src/ActionHeader.jsx:75]: Happy to see this convention spreading!

File: packages/terra-action-header/src/ActionHeader.jsx
Comment (xenoworf) [inline packages/terra-action-header/src/ActionHeader.jsx:92]: Why must a required prop specify a default value?

Comment (saket2403) [reply packages/terra-action-header/src/ActionHeader.jsx:92]: It is not a required prop.

File: packages/terra-action-header/src/ActionHeader.jsx
Comment (saket2403) [reply packages/terra-action-header/src/ActionHeader.jsx:237]: Yes the visually-hidden-text has a width of 1px which is causing a slight shift.

File: packages/terra-action-header/src/ActionHeader.jsx
Comment (xenoworf) [inline packages/terra-action-header/src/ActionHeader.jsx:121]: Please use the AccessibleValue component, which I think will solve this double-reading problem and was purpose built for this kind of scenario: https://github.com/cerner/terra-framework/blob/c92fd06dc900c13f4c3ce8457066c7db15a5a2f6/packages/terra-time-input/src/_AccessibleValue.jsx#L25

That component was created as part of adding a11yLabels for TimeInput.

**Review by xenoworf — CHANGES_REQUESTED** [review]
Requesting to use AccessibleValue component. Other than that this code looks fine.

File: packages/terra-action-header/src/ActionHeader.jsx
Comment (benbcai) [reply packages/terra-action-header/src/ActionHeader.jsx:237]: Is this approved by UX? Otherwise, I think we need to adjust the styles to make it look how it used to be.

One possible option is to add the appropriate padding/margin to this instance of `visuallyHiddenComponent` to match how it used to be.

File: packages/terra-action-header/src/ActionHeader.jsx
Comment (xenoworf) [reply packages/terra-action-header/src/ActionHeader.jsx:121]: If you'd prefer, perhaps ship-as-is and then go back and use this component? I think that would be passive and might help you ship faster. Your call. If you'd like to do the iterative approach, tell me so and I'll flip my Request Change to Approved :).

Comment (saket2403) [reply packages/terra-action-header/src/ActionHeader.jsx:121]: Yes I think that would be better, also if the AccessibleValue is a reusable component for these scenarios, I think it would be better to move it to a common folder instead of importing it from time input or adding the same code for using it in another component.

Comment (xenoworf) [reply packages/terra-action-header/src/ActionHeader.jsx:121]: Agreed. A third option is to use AccessibleValue as-is, and then move it later - which would still be passive. 

So I take it you will be shipping as-is and then turning around to use AccessibleValue in some form?

Comment (saket2403) [reply packages/terra-action-header/src/ActionHeader.jsx:121]: Yes. For the time being I will be shipping as is and then change to AccessibleValue at a later stage.

**Review by xenoworf — APPROVED** [review]

File: packages/terra-action-header/src/ActionHeader.jsx
Comment (neilpfeiffer) [reply packages/terra-action-header/src/ActionHeader.jsx:237]: Putting the visually hidden content in the startContent is the wrong location to insert. By including it here is it what is causing the current screenshot changes and the look is undesired UX changes. 

We intentionally created different left margin alignments depending on if there are action-header buttons or not in the start content, shifting between 10px if no contents, to 5px if buttons are present (has to do with how the no border buttons visually look to best align). The css that applies these differences is here: https://github.com/cerner/terra-core/blob/a7c88211833e8f180dec97341ef28bbdc590355e/packages/terra-action-header/src/ActionHeaderContainer.module.scss#L69

I would recommend inserting the visually hidden components within their own 0 height parent row, that will render on its own row below the entire current action-header contents flex row.

**Review by neilpfeiffer — DISMISSED** [review]
Requesting changes to where the visually hidden content is inserted in order to restore correct visual screenshots: [see comment for ActionHeader.jsx: line 237](https://github.com/cerner/terra-core/pull/3645/files?file-filters%5B%5D=.jsx&file-filters%5B%5D=.snap&show-viewed-files=true#r935064769)

**Pull request merged**
